### PR TITLE
chore: update tap@19

### PR DESCRIPTION
**Pull request by climba03003**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jsumners** [comment]
Unless things have changed, it's a little bit more involved https://github.com/fastify/fastify/pull/4978/files

**Comment by climba03003** [comment]
> Unless things have changed

Have tested locally, it only requires minimal changes.
Is it hard requirement to remove `c8`?

**Comment by jsumners** [comment]
`tap@18` is using c8 itself.

**Comment by climba03003** [comment]
After hanging a bit on removing the external `c8` and using the internal one.
It forced to includes the files that should be ignored in the coverage report.

**Review by gurgunday — APPROVED** [review]

**Comment by gurgunday** [comment]
Are macos tests flaky?

**Review by fox1t — APPROVED** [review]

**Review by gurgunday — COMMENTED** [review]
Seems like only macOS fails with a heap limit error

**Comment by gurgunday** [comment]
Do you know why macOS unit test CI could be failing?

**Comment by Cangit** [comment]
Test runs out of memory. Its because the test relied on a regression in node v18 that was recently fixed in node v18.9:

`This release fixes a regression introduced in Node.js 18.19.0 where http.server.close() was incorrectly closing idle connections.`

I have updated the test, and also updated to tap19 that was released after this PR was initially opened. 
(Opened as new PR because restrictions) #5495

**Review by gurgunday — CHANGES_REQUESTED** [review]
Can you update to tap@19 as well?

File: package.json [hunk 1]
```diff
@@ -11,24 +11,24 @@
     "benchmark:parser": "concurrently -k -s first \"node ./examples/benchmark/parser.js\" \"autocannon -c 100 -d 30 -p 10 -i ./examples/benchmark/body.json -H \"content-type:application/jsoff\" -m POST localhost:3000/\"",
     "build:validation": "node build/build-error-serializer.js && node build/build-validation.js",
     "coverage": "npm run unit -- --coverage-report=html",
-    "coverage:ci": "c8 --reporter=lcov tap --coverage-report=html --no-browser --no-check-coverage",
-    "coverage:ci-check-coverage": "c8 check-coverage --branches 100 --functions 100 --lines 100 --statements 100",
+    "coverage:ci": "tap --coverage-report=html --coverage-report=lcov --allow-incomplete-coverage",
+    "coverage:ci-check-coverage": "tap replay",
```
Comment (climba03003) [inline package.json:15]: Replay commands will fail if threshold is not meet
See https://github.com/fastify/fastify/actions/runs/9362073855 for demostration

File: lib/configValidator.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // This file is autogenerated by build/build-validation.js, do not edit
 /* istanbul ignore file */
```
Comment (gurgunday) [inline lib/configValidator.js:2]:
```suggestion
```

File: build/build-error-serializer.js [hunk 3]
```diff
@@ -19,9 +19,12 @@ const file = path.join(__dirname, '..', 'lib', 'error-serializer.js')
 
 const moduleCode = `// This file is autogenerated by build/build-error-serializer.js, do not edit
 /* istanbul ignore file */
```
Comment (gurgunday) [inline build/build-error-serializer.js:21]:
```suggestion
```

File: build/build-validation.js [hunk 4]
```diff
@@ -10,9 +10,11 @@ const factory = AjvStandaloneCompiler({
   storeFunction (routeOpts, schemaValidationCode) {
     const moduleCode = `// This file is autogenerated by ${__filename.replace(__dirname, 'build')}, do not edit
 /* istanbul ignore file */
```
Comment (gurgunday) [inline build/build-validation.js:12]:
```suggestion
```

File: lib/error-serializer.js [hunk 5]
```diff
@@ -1,11 +1,10 @@
 // This file is autogenerated by build/build-error-serializer.js, do not edit
 /* istanbul ignore file */
```
Comment (gurgunday) [inline lib/error-serializer.js:2]:
```suggestion
```

**Review by gurgunday — COMMENTED** [review]
`istanbul-ignore`s don't do anything anymore no?

Please make sure to remove them

File: lib/error-serializer.js [hunk 6]
```diff
@@ -23,40 +33,88 @@
...
+            json += JSON_STR_QUOTE + value.toISOString() + JSON_STR_QUOTE
+          } else if (value instanceof RegExp) {
+            json += serializer.asString(value.source)
+          } else {
+            json += serializer.asString(value.toString())
+          }
+        } else {
+          json += serializer.asString(value)
+        }
+
```
[the remark this answers is not in the document]
Comment (climba03003) [reply lib/error-serializer.js:67]: Auto generated file is not for review.

**Review by gurgunday — APPROVED** [review]
Looks great, thanks!

**Pull request merged**